detectors/TransLab/network_runner.py

```diff
@@ -0,0 +1,145 @@
+)
+from segmentron.config import cfg
+from segmentron.utils.default_setup import default_setup
+
+
+class NetworkRunner(NetworkRunnerBase):
+    def __init__(
+        self,
+        input_dir: Path,
+        output_dir: Path,
+        log_path: Path,
+        model_path: Path,
+        segmentron_args,
+    ):
+        root_path = os.path.abspath(os.path.dirname(__file__))
+        sys.path.append(root_path)
+
+        cfg.TEST.TEST_MODEL_PATH = str(model_path)
+        cfg.DEMO_DIR = str(input_dir)
+        cfg.update_from_file("configs/trans10K/translab.yaml")
+        cfg.PHASE = "test"
+        cfg.ROOT_PATH = root_path
+        cfg.DATASET.NAME = "trans10k_extra"
+        cfg.check_and_freeze()
+
+        sys.argv = [__file__] + segmentron_args
+        args = options.parse_args()
+        default_setup(args)
+
+        self.args = args
+        self.device = torch.device(args.device)
+
+        super().__init__(input_dir, output_dir, log_path, model_path)
+
+        self._init_dataset()
+
+    def _image_gen(self) -> Tuple[Any, Path, Tuple[int, int]]:
+        for img, _, img_name in self.val_loader:
+            img_path = Path(img_name[0])
+            ori_img = cv2.imread(str(img_path))
+            h, w, _ = ori_img.shape
+            yield img, img_path, (w, h)
+
+    def _predict(self, img, shape):
+        with torch.no_grad():
+            output, output_boundary = self.model.evaluate(img)
+
+            glass_res = output.argmax(1)[0].data.cpu().numpy().astype("uint8") * 127
+            glass_res = cv2.resize(glass_res, shape, interpolation=cv2.INTER_NEAREST)
+        return glass_res
+
+    def _write_img(self, img_name, prediction):
+        save_path = self.output_dir
+        cv2.imwrite(os.path.join(save_path, img_name), prediction)
+
+    def _init_dataset(self):
+        input_dir = self.input_dir
+        # image transform
+        input_transform = transforms.Compose(
+            [
+                transforms.ToTensor(),
+                transforms.Normalize(cfg.DATASET.MEAN, cfg.DATASET.STD),
+            ]
+        )
+
+        # dataset and dataloader
+        val_dataset = get_segmentation_dataset(
+            cfg.DATASET.NAME,
+            root=input_dir,
+            split="val",
+            mode="val",
+            transform=input_transform,
+            base_size=cfg.TRAIN.BASE_SIZE,
+        )
+
+        val_sampler = make_data_sampler(
+            val_dataset, shuffle=False, distributed=self.args.distributed
+        )
+        val_batch_sampler = make_batch_data_sampler(
+            val_sampler, images_per_batch=cfg.TEST.BATCH_SIZE, drop_last=False
+        )
+
+        self.val_loader = data.DataLoader(
+            dataset=val_dataset,
+            batch_sampler=val_batch_sampler,
+            num_workers=cfg.DATASET.WORKERS,
+            pin_memory=True,
+        )
+        self.classes = val_dataset.classes
+
+    def _set_batch_norm_attr(self, named_modules, attr, value):
```

Comment:
Please use two underscores `__` before the name if a method is not expected to be overridden